### PR TITLE
Patient tools tests

### PR DESCRIPTION
**Pull request by tmarathe**

### CODE REVIEW

**Comment by tmarathe** [comment]
Nominating @DeanWay, @antonkatz for review.

File: components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+         testGroup = new FormGroup("Test Group");
+         fieldnames = new String[] { "phenotype", "negative_phenotype" };
+         fieldnamesWithoutNO = new String[] { "phenotype" };
+
+     }
+
+    @Test
+    public void checkElementsAreAdded(){
+        FormField testNullField = null;
+        Assert.assertEquals(false, testGroup.addElement(testNullField));
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java]: Remove this variable -> Assert.assertEquals(false, testGroup.addElement(null);

File: components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        FormField testField = mock(FormField.class);
+        Assert.assertEquals(true, testGroup.addElement(testField));
+        Assert.assertTrue(testGroup.elements.contains(testField));
+    }
+
+    @Test
+    public void checkEmptyFormGroupDisplay(){
+        Assert.assertEquals(testGroup.display(DisplayMode.Edit, new String[] { "phenotype", "negative_phenotype" }), "");
+        Assert.assertEquals(testGroup.display(DisplayMode.View, new String[] { "phenotype", "negative_phenotype" }), "");
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java]: Format for assertions is `Assert.assertSomthing(EXPECTED, ACTUAL)`.
Change the order here

File: components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public void addAndGetCustomElements(){
+        Assert.assertNotNull(testFormSection.addCustomElement(testFormField));
+        Assert.assertNotNull(testFormSection.addCustomElement(testFormGroup));
+        Assert.assertNotNull(testFormSection.getCustomElements());
+    }
+
+    @Test
+    public void displaySection() {
+        Assert.assertNotEquals(testFormSection.display(DisplayMode.Edit, fieldNames), "");
+        Assert.assertEquals(testFormSection.display(DisplayMode.View, fieldNames), "");
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java]: Again here, `assertSomething(EXPECTED, ACTUAL)`

File: components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        testFormSection.addElement(testFormGroup);
+        String expectedDisplay = "<div class='phenotype-group' style=''><h3 id='Htitle'><span>" + title + "</span></h3>"
+            + "<div class='phenotype-main predefined-entries'>null</div><div class='phenotype-other custom-entries'>"
+            + "<div class=\"custom-display-data\">null</div><label for='phenotype_0." + "\\d+" + "' class='label-other "
+            + "label-other-phenotype'>Other</label><input type='text' name='phenotype' class='suggested multi suggest-hpo "
+            + "generateYesNo accept-value' value='' size='16' id='phenotype_0." + "\\d+" + "' placeholder='enter free "
+            + "text and choose among suggested ontology terms'/><input type='hidden' value='' name='_category'/></div>"
+            + "</div>";
+        String displayResult = testFormSection.display(DisplayMode.Edit, fieldNames);
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java]: `Displaymode.View` is not tested for FormSection display()

File: components/patient-tools/src/test/java/org/phenotips/tools/FormSubsectionTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        String type = "type";
+
+        FormSubsection testSubsection = new FormSubsection(title, type);
+        Assert.assertEquals(testSubsection.display(
+            DisplayMode.Edit, new String[]{ "phenotype", "negative_phenotype" }), "");
+        FormField testFormField = mock(FormField.class);
+        testSubsection.addElement(testFormField);
+        Assert.assertEquals(testSubsection.display(
+            DisplayMode.Edit, new String[]{"phenotype", "negative_phenotype"}),
+            "<label class='section'>" + title + "</label><div class='subsection "+ type + "'>null</div>");
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormSubsectionTest.java]: Here as well

File: components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java (hunk 1 above)
Comment (sdumitriu) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java]: Or better: `Assert.assertFalse(testGroup.addElement(null));`

File: components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        testFormField = mock(FormField.class);
+        testFormGroup = mock(FormGroup.class);
+        fieldNames = new String[]{"phenotype", "negative_phenotype"};
+    }
+
+    @Test
+    public void addAndGetCustomElements(){
+        Assert.assertNotNull(testFormSection.addCustomElement(testFormField));
+        Assert.assertNotNull(testFormSection.addCustomElement(testFormGroup));
+        Assert.assertNotNull(testFormSection.getCustomElements());
```
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java]: Perhaps test the `setCustomElements()` method here as well

File: components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java (hunk 4 above)
Comment (tmarathe) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormSectionTest.java]: @DeanWay It is tested on line 58. FormSection display() should always return "" in DisplayMode.View.

File: components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java (hunk 1 above)
Comment (DeanWay) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormGroupTest.java]: Right, there are also instances in this pull request where `assertEquals()` is used for boolean return values that should be replaced be `assertTrue()` or `assertFalse()`

**Comment by DeanWay** [comment]
:+1:

**Comment by sdumitriu** [comment]
@antonkatz Your review was requested.

File: components/patient-tools/src/test/java/org/phenotips/tools/FormConditionalSubsectionTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        FormConditionalSubsection testConditionalSubsection = new FormConditionalSubsection(
+            title, type, testYesNoPicker, false);
+        Assert.assertEquals("", testConditionalSubsection.display(
+            DisplayMode.Edit, new String[]{"phenotypes", "negative_phenotypes"}));
+        Assert.assertEquals("", testConditionalSubsection.display(
+            DisplayMode.View, new String[]{"phenotypes", "negative_phenotypes"}));
+    }
+
+    @Test
+    public void SelectedDisplay(){
```
Comment (sdumitriu) [inline components/patient-tools/src/test/java/org/phenotips/tools/FormConditionalSubsectionTest.java]: Method names must start with a lowercase letter.

**Pull request merged**